addon/-private/system/store.js

```diff
@@ -2672,7 +2699,7 @@ function _commit(adapter, store, operation, snapshot) {
       if (adapterPayload) {
         payload = normalizeResponseHelper(serializer, store, modelClass, adapterPayload, snapshot.id, operation);
         if (payload.included) {
-          store.push({ data: payload.included });
+          store._push({ data: null, included: payload.included });
```

Comment:
@bcardarella the main difference is that it doesn't materialize the records, however it also will likely become the basis for a new public API that's currently going through the RFC process.